Library/Homebrew/extend/os/mac/requirements/java_requirement.rb

```diff
@@ -13,6 +13,7 @@ def possible_javas
     javas << Pathname.new(ENV["JAVA_HOME"])/"bin/java" if ENV["JAVA_HOME"]
     javas << java_home_cmd
     javas << which("java")
+    javas.delete(Pathname.new("/usr/bin/java")) # /usr/bin/java is a stub on macOS
```

Comment:
I agree with this change. However, I think we should also return `javas.compact` instead of `javas`, since `java_home_cmd` can return `nil` and will likely end up first in the array for many users.